api/cloud/oc_cloud_manager.c

```diff
@@ -458,17 +462,22 @@ cloud_login_handler(oc_client_response_t *data)
       reset_delayed_callback(ctx, refresh_token,
                              check_expires_in(ctx->store.expires_in));
     }
-  } else {
-    if (data->code != OC_STATUS_UNAUTHORIZED) {
-      oc_set_delayed_callback(ctx, cloud_login,
-                              g_retry_timeout[ctx->retry_count]);
-    } else {
-      if (oc_string(ctx->store.refresh_token) &&
-          oc_string_len(ctx->store.refresh_token) > 0) {
-        reset_delayed_callback(ctx, refresh_token, g_retry_timeout[0]);
-      }
+    goto finish;
+  }
+  if (data->code == OC_STATUS_UNAUTHORIZED) {
+    if (oc_string(ctx->store.refresh_token) &&
+        oc_string_len(ctx->store.refresh_token) > 0) {
+      ctx->store.status &= ~OC_CLOUD_LOGGED_IN;
+      ctx->store.status |= OC_CLOUD_UNAUTHORIZED;
```

Comment:
I'm not sure if this line -> because if interface goes down cloud manager is not restarted.